### ISSUE TITLE
Release 4.5.0 breaking tests

### ISSUE THREAD

**Issue by selimt** [pytest-dev/pytest-django#975]
Since 4.5.0 is released, our tests that rely on the loaddata functionality is breaking. Here is our fixture:

```
@pytest.fixture(scope='session')
def django_db_setup(django_db_setup, django_db_blocker):
    """Fixture for initializing the database with test data."""
    # This will populate the database with the data in the test_data.json file
    # That file was initially created by running a manage.py dumpdata on a test database
    # with some relevant testing data
    with django_db_blocker.unblock():
        call_command('loaddata', 'test_data.json')
```

If we revert to 4.4.0, then all tests pass. We are using Python 3.7 and Django 3.2.9.

Thanks

**Comment by bluetech** [comment]
How does it break?

**Comment by selimt** [comment]
I haven't investigated it enough but basically the tests don't find the expected data from the server. Here is some initial output from the test session:

```
============================= test session starts ==============================
platform linux -- Python 3.7.7, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
django: settings: catalog_server.settings.settings_test_catalog_api (from env)
rootdir: /, configfile: pytest.ini
plugins: cov-3.0.0, django-4.5.0, localserver-0.4.1, mock-3.6.1
collected 173 items

../../../../tests/catalog_api/test_model.py .                            [  0%]
../../../../tests/catalog_api/test_api.py .............................F [ 17%]
FFFFFFF..FFFF.FF...F.FFFFFFFFFF.F.FF.FF.FFFF.FFFF...............         [ 54%]
../../../../tests/catalog_api/test_model.py ...                          [ 56%]
../../../../tests/catalog_misc/test_template.py .                        [ 57%]
../../../../tests/catalog_search/test_api.py ..............FF........... [ 72%]
...F..FFF.........FFF....                                                [ 87%]
../../../../tests/catalog_server/test_login.py ...                       [ 89%]
../../../../tests/catalog_sync/test_sync.py ..................s          [100%]

=================================== FAILURES ===================================
_____________________ TestTaggables.test_filter_taggables ______________________

self = <tests.catalog_api.test_api.TestTaggables object at 0x7fe86efba710>
req_helper = <function req_helper.<locals>.req_helper_func at 0x7fe86eb9da70>
db_data = [{'fields': {'metadata': {'_system': {'created': '2021-03-02T23:03:32.332954+00:00', 'created_by': '####'}, 'show...fa4-9138-43d7-af96-43955f38e7cc'}, 'model': 'catalog_api.taggable', 'pk': '05fb6c08-0731-4387-945b-a5fdd117422e'}, ...]

    def test_filter_taggables(self, req_helper, db_data):
        """Test search filter for taggables."""
        key = 'name'
        value = 'dog'
        resp = req_helper('GET', format_endpoint('taggables/?{}={}'.format(key, value)))
        expected_paths = [d for d in db_data if d['model'] == CATALOG_API_TAGGABLE
                          and d['fields'][key] == value]
>       assert resp.json()['count'] == len(expected_paths)
E       AssertionError: assert 0 == 1
E        +  where 1 = len([{'fields': {'accessor': {'id': '1234'}, 'metadata': {'_system': {'created': '2021-03-02T23:07:11.257818+00:00', 'crea...a0a26fdb-1f57-4fd3-b5cc-1d04ae61ae88'}, 'model': 'catalog_api.taggable', 'pk': '981b4dc3-4c2b-4aeb-9259-4a21b72a16d0'}])

/tests/catalog_api/test_api.py:472: AssertionError
```

**Comment by bluetech** [comment]
I have a big project which uses this pattern (`loaddata` in `django_db_setup`) and everything works as intended there. So it's not something obvious.

- Does the loaddata work at all? That is, is the data loaded e.g. in the first test and then disappears?
- Are you using transactional tests (`pytest.mark.django_db(transaction=True)` or `transactional_db` fixture)? These tests flush out the DB.

**Comment by selimt** [comment]
We do have transactional tests but those are not the tests that are breaking.

I also tried running the tests that are failing individually (using the -k flag) and they pass. So it must be some kind of precedence issue or a combination of conftest.py modules when running all the tests.

We are also running the tests within a docker container, not sure if that matters.

Let me dig into it more and see where exactly it is breaking.

**Comment by bluetech** [comment]
Transactional tests flush the DB, so they clear all of the data that you load in your `django_db_setup` fixture.

pytest-django makes an effort to reorder tests such that transactional tests always run *after* regular tests. Maybe this is where the bug is. So one thing to check is to compare the output of `pytest --collect-only` in pytest-django 4.4 vs. 4.5.

**Comment by selimt** [comment]
Yes I think you are right. Comparing the "--collect-only" output, I see that there are some transactional tests that execute at the end for 4.4.0 but not for 4.5.0.

**Comment by bluetech** [comment]
Should be fixed in 4.5.1, thanks for the bug report.
